server/expression/data_store.go

```diff
@@ -21,7 +22,12 @@ func (ds AttributeDataStore) Source() string {
 }
 
 func (ds AttributeDataStore) Get(name string) (string, error) {
-	return ds.Span.Attributes.Get(name), nil
+	value := ds.Span.Attributes.Get(name)
+	if value == "" {
+		return "", errors.New("attribute not found")
```

Comment:
Does it make sense to type this error and send the attribute name as metadata? To handle later on the events or the API?

Something like this:

```go
// somewhere in the code
type ErrAttributeNoFound struct {
    attributeName string
}

func (e *ErrAttributeNoFound) Error() string {
    return fmt.Sprintf("attribute %s not found", e.attributeName)
} 

func (e *ErrAttributeNoFound) AttributeName() string {
    return e.attributeName
} 

// here
return "", ErrAttributeNoFound{attributeName: name}

```

Reply:
I added part of your idea to the `ResolutionError` object. While reading the errors.Is() source code, I saw that you can declare the `Is` and `Unwrap` functions to a struct and the `errors` package will be able to use those custom-defined functions as part of the standard `Is` and `Unwrap` functions.